packages/svelte/src/internal/client/reactivity/batch.js

```diff
@@ -829,6 +872,71 @@ export function eager(fn) {
 	return value;
 }
 
+/**
+ * @param {() => void} fn
+ * @returns {{ commit: () => void, discard: () => void }}
+ */
+export function fork(fn) {
+	if (!async_mode_flag) {
+		e.experimental_async_fork();
+	}
+
+	if (current_batch !== null) {
+		throw new Error('cannot fork here'); // TODO better error
+	}
+
+	const batch = Batch.ensure();
+	batch.is_fork = true;
+
+	const settled = batch.settled();
+
+	flushSync(fn);
+
+	// revert state changes
+	for (const [source, value] of batch.previous) {
+		source.v = value;
+	}
+
+	return {
+		commit: async () => {
+			if (!batches.has(batch)) {
+				throw new Error('Cannot commit this batch'); // TODO better error
+			}
+
+			batch.is_fork = false;
+
+			// delete all other forks
```

Comment:
This makes me a bit nervous with respects to usage. How can a user know that there's no other fork? Maybe I have a tiny fork for one part of the UI and a separate one for the other. As long as they don't overlap should we keep them? AFAICT transitions in other frameworks (the closest to our forks) can also coexist.

Reply:
Hmm, maybe it's okay for this particular knife to have a sharp edge, given that it's meant for framework authors/power users